assets/app/view/game/actionable.rb

```diff
@@ -32,11 +32,15 @@ def process_action(action)
         end
 
         if !hotseat &&
-            !action.free? &&
-            participant &&
-            !@game.active_players.map(&:name).include?(@user['name']) &&
-            !Lib::Storage[@game.id]&.dig('master_mode')
-          return store(:flash_opts, 'Not your turn. Turn on master mode in the tools tab to act for others.')
+          !action.free? &&
+          participant &&
+          !@game.active_players.map(&:name).include?(@user['name'])
+
+          unless Lib::Storage[@game.id]&.dig('master_mode')
+            return store(:flash_opts, 'Not your turn. Turn on master mode in the tools tab to act for others.')
+          end
+
+          action.master_user = @user['name']
```

Comment:
i don't think this is needed, we already know which user made every action in the database

Reply:
Without this code, the user that made the master mode call will not see the master mode log until new data is returned from the server (either refresh or other player action), as _@game.process_action_ (line 42 of actionable) is run on the client, where there is no user data yet.

It is one line to add the user data server side, and even with this code I would suggest adding it as it is a more "reliable" method for determining if it is a master mode action, but I don't see how we can do without the above code unless we just don't care about the acting user see the log.

I've added that line to the PR, but I am interested if there is something obvious I am missing.

Reply:
action.master_user needs to change to user

Reply:
Did I convince you we should keep this then? :D

Reply:
yes it's fine, thanks for the explanation